libs/components/indicators/testing/src/wait/wait-harness.spec.ts

```diff
@@ -0,0 +1,133 @@
+@Component({
+  selector: 'sky-wait-test',
+  template: `
+    <div style="width: 200px; height: 100px">
+      Sample Component
+      <sky-wait
+        [ariaLabel]="ariaLabel"
+        [isFullPage]="isFullPage"
+        [isNonBlocking]="isNonBlocking"
+        [isWaiting]="isWaiting"
+        data-sky-id="test-wait"
+      ></sky-wait>
+    </div>
+    <div style="width: 200px; height: 100px">
+      <sky-wait
+        [ariaLabel]="ariaLabel"
+        [isFullPage]="isFullPage"
+        [isNonBlocking]="isNonBlocking"
+        [isWaiting]="isWaiting"
+        data-sky-id="wait-2"
+      ></sky-wait>
+    </div>
+  `,
+})
+class TestComponent {
+  public ariaLabel: string | undefined;
+
+  public isFullPage: boolean | undefined;
+
+  public isNonBlocking: boolean | undefined;
+
+  public isWaiting = false;
+}
+//#endregion Test component
+
+async function validateWaitProperties(
+  waitHarness: SkyWaitHarness,
+  fixture: ComponentFixture<TestComponent>,
+  isFullPage: boolean,
+  isNonBlocking: boolean,
+  ariaLabel?: string
+): Promise<void> {
+  fixture.componentInstance.ariaLabel = undefined;
+
+  fixture.componentInstance.isFullPage = isFullPage;
+  fixture.componentInstance.isNonBlocking = isNonBlocking;
+
+  if (ariaLabel) {
+    fixture.componentInstance.ariaLabel = ariaLabel;
+  }
+
+  fixture.componentInstance.isWaiting = true;
+  fixture.detectChanges();
+
+  const label =
+    fixture.componentInstance.ariaLabel ??
+    `${fixture.componentInstance.isFullPage ? 'Page l' : 'L'}oading.${
+      fixture.componentInstance.isNonBlocking ? '' : ' Please wait.'
+    }`;
+  await expectAsync(waitHarness.getAriaLabel()).toBeResolvedTo(label);
+  await expectAsync(waitHarness.isWaiting()).toBeResolvedTo(true);
+  await expectAsync(waitHarness.isFullPage()).toBeResolvedTo(isFullPage);
+  await expectAsync(waitHarness.isNonBlocking()).toBeResolvedTo(isNonBlocking);
+  fixture.componentInstance.isWaiting = false;
+}
+
+describe('Wait harness', () => {
+  async function setupTest(
+    options: { dataSkyId?: string; ariaLabel?: string } = {}
+  ): Promise<{
+    waitHarness: SkyWaitHarness;
+    fixture: ComponentFixture<TestComponent>;
+    loader: HarnessLoader;
+  }> {
+    await TestBed.configureTestingModule({
+      declarations: [TestComponent],
+      imports: [SkyWaitModule],
+    }).compileComponents();
+
+    const fixture = TestBed.createComponent(TestComponent);
+    const loader = TestbedHarnessEnvironment.loader(fixture);
+
+    let waitHarness: SkyWaitHarness;
+
+    if (options.dataSkyId) {
+      waitHarness = await loader.getHarness(
+        SkyWaitHarness.with({
+          dataSkyId: options.dataSkyId,
+        })
+      );
+    } else {
+      waitHarness = await loader.getHarness(SkyWaitHarness);
+    }
+
+    if (options.ariaLabel) {
+      fixture.componentInstance.ariaLabel = options.ariaLabel;
+    }
+
+    return { waitHarness, fixture, loader };
+  }
+
+  it('should return the expected ARIA label', async () => {
```

Comment:
This description doesn't match the list of expectations, below. Maybe separate all of these into separate `it` block to confirm the harness checks for full-page, blocking, ARIA label, etc.?

Reply:
I like the conciseness of leaving them all in one method. Updated the test description to make it clear it's exercising all of the harness properties